### PR TITLE
Ensure gutter icons show up in the correct locations

### PR DESCRIPTION
**Pull request by keyboardDrummer**
Fixes https://github.com/dafny-lang/ide-vscode/issues/237
Fixes https://github.com/dafny-lang/ide-vscode/issues/235

### Changes
- Add a version check to prevent processing out-of-date symbolStatus updates
- Enable turning verification status in the test panel on/off

### Caveats
If you hold the enter key to continuously insert newlines, the 'play' icon from the verification symbol status still lags behind the moving text. I've tried to find whether it applies outdated data, but everything checks out, so I'm very confused as to while the visuals don't.

### CODE REVIEW

**Comment by MikaelMayer** [comment]
> Fixes #237

Thanks for trying ! However, I just tested this PR, I have the same problem (I pressed three times ENTER, this is what I got the first time)

image

**Comment by keyboardDrummer** [comment]
> Thanks for trying ! However, I just tested this PR, I have the same problem (I pressed three times ENTER, this is what I got the first time)

I wasn't able to reproduce that, but I also made some more changes to improve the behavior of the play icon.

**Comment by MikaelMayer** [comment]
> I wasn't able to reproduce that, but I also made some more changes to improve the behavior of the play icon.

Just tried your new version, this bug seems indeed harder to obtain, but by writing two methods, and long-pressing and short-pressing enter, I get similar overlaps
image

**Comment by MikaelMayer** [comment]
> > I wasn't able to reproduce that, but I also made some more changes to improve the behavior of the play icon.
> 
> Just tried your new version, this bug seems indeed harder to obtain, but by writing two methods, and long-pressing and short-pressing enter, I get similar overlaps image

What is worse than that, is that an incorrect state in the language server seems to persist when I create a new unrelated document, I get this:

image

**Review by MikaelMayer — CHANGES_REQUESTED** [review]
A few changes, otherwise, looks good ! I wasn't able to reproduce the bugs that this PR claims to fix, great job.

File: src/ui/verificationSymbolStatusView.ts [hunk 1]
```diff
@@ -78,18 +42,28 @@ export default class VerificationSymbolStatusView {
...
       const uriString = e.document.uri.toString();
       this.updateListenersPerFile.delete(uriString);
-    });
+      this.updatesPerFile.delete(uriString);
+    }, this, context.subscriptions);
     workspace.onDidCloseTextDocument(e => {
       const uriString = e.uri.toString();
       this.updatesPerFile.delete(uriString);
-    });
+    }, this, context.subscriptions);
```
[the remark this answers is not in the document]
Comment (keyboardDrummer) [reply src/ui/verificationSymbolStatusView.ts:62]: `workspace.onDidCloseTextDocument` creates a disposable, and without passing these arguments that disposable is never disposed.

**Review by MikaelMayer — APPROVED** [review]
Thank you for making this extension configurable. Since it's now integrated in the back-end though, I suspect deactivating it won't prevent any errors though.

**Pull request merged**